### PR TITLE
Handling ogg format

### PR DESCRIPTION
**Pull request by Wameuh**
This version is not the best possible because it extracts the audio file before decoding it.

### CODE REVIEW

File: pkg/lunii/writer.go [hunk 1]
```diff
@@ -202,6 +203,10 @@ func convertAndWriteAudio(reader zip.ReadCloser, deviceAudioDirectory string, au
 		mp3, _ = hex.DecodeString(BLANK_MP3_FILE)
 
 	} else {
+		start := time.Now()
+
+		//check extension
+		extension := audio.SourceName[len(audio.SourceName)-3:]
```
Comment (olup) [inline pkg/lunii/writer.go:209]: There is a golang utils for this : `path.Ext` I think

File: pkg/lunii/writer.go [hunk 2]
```diff
@@ -210,15 +215,49 @@ func convertAndWriteAudio(reader zip.ReadCloser, deviceAudioDirectory string, au
...
+			}
+			mp3, err = Mp3ToMp3(fileBytes)
+			if err != nil {
+				return err
+			}
+		} else if extension == "ogg" {
+			// it's an ogg file
+
+			// Didn't work for the moment with the file open with the zip.ReadCloser. So save a temp file and use it.
+			tmpFile, err := os.Create("tmp_" + audio.SourceName)
```
Comment (olup) [inline pkg/lunii/writer.go:232]: There is golang utils to get the tmp directory of the system an create a tmp file, maybe you should use this ?

File: pkg/lunii/writer.go [hunk 3]
```diff
@@ -210,15 +215,49 @@ func convertAndWriteAudio(reader zip.ReadCloser, deviceAudioDirectory string, au
...
+			}
+
+			tmpFile.Close()
+			tmpFile, err = os.Open("tmp_" + audio.SourceName)
+			if err != nil {
+				return err
+			}
+
+			defer tmpFile.Close()
+			defer os.Remove("tmp_" + audio.SourceName)
```
Comment (olup) [inline pkg/lunii/writer.go:249]: I think you should put this one right after creating the file l.232, it would be clearer

**Comment by olup** [comment]
Hey, so cool ! Thanks for taking the time. Left a few comments around

**Review by olup — APPROVED** [review]

**Comment by olup** [comment]
Meged ! Now we need to update lunii-admin with the latest version of the cli. Future improvement: now that you unzip first, we might consider unzipping everything beforehand first. It might be more performant (or might not, not sure)

**Comment by Wameuh** [comment]
> Meged ! Now we need to update lunii-admin with the latest version of the cli. Future improvement: now that you unzip first, we might consider unzipping everything beforehand first. It might be more performant (or might not, not sure)

I am not sure that will be more performant, because images will be unzipped and written on the disk. Also, if the files are mp3, they will be unzipped even if it's not necessary.

**Pull request merged**